syntax = "proto3";

package ck.game;

import "common_entities.proto";

import "game/entities.proto";

option csharp_namespace = "Services.Game.Protobuf";

option java_package = "com.devsisters.ck.services.game.protobuf";

message GetMailBoxRequest {
}

message GetMailBoxResponse {
  MailBox mail_box = 1;
}

message GetMailBoxSummaryRequest {
}

message GetMailBoxSummaryResponse {
  MailBoxSummary mail_box_summary = 1;
}

message ReceiveMailRewardsRequest {
  repeated string mail_ids = 1;
}

message ReceiveMailRewardsResponse {
  KingdomAdditionalEvents additional_events = 1;

  repeated Mail updated_mails = 2;

  repeated Reward rewards = 3;
}

service MailService {
  rpc GetMailBox ( GetMailBoxRequest ) returns ( GetMailBoxResponse );

  rpc GetMailBoxSummary ( GetMailBoxSummaryRequest ) returns ( GetMailBoxSummaryResponse );

  rpc ReceiveMailRewards ( ReceiveMailRewardsRequest ) returns ( ReceiveMailRewardsResponse );
}
